File: python/n367_SqrNum.py
# https://leetcode-cn.com/problems/valid-perfect-square/

class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num == 1: return True
        left = 0
        right = num // 2       
        while left <= right:
            mid = int((left + right)/2)
            if mid * mid == num:
                return True
            elif mid * mid < num:
                left = mid + 1               
            else:
                right = mid - 1
        return False
        

if __name__ == "__main__":
    solu = Solution()
    print(solu.isPerfectSquare(256))

File: cpp/n367_SqrNum.cpp
// https://leetcode-cn.com/problems/valid-perfect-square/
#include <iostream>
#include <vector>
#include <sstream>
#include <string>
#include <algorithm>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool isPerfectSquare(int num) {
        int minu = 1;
        while (num > 0){
            num -= minu;
            minu += 2;
        }
        return num == 0;
    }
};

int main(){
    Solution solu;
    cout << solu.isPerfectSquare(25) << endl;
}